src/diffusers/models/autoencoders/autoencoder_dc.py

```diff
@@ -0,0 +1,626 @@
+        key = self.kernel_nonlinearity(key)
+        value = F.pad(value, (0, 0, 0, 1), mode="constant", value=1)
+
+        key_T = key.transpose(-1, -2)
+        scores = torch.matmul(value, key_T)
+        output = torch.matmul(scores, query)
+
+        output = output.float()
+        output = output[:, :, :-1] / (output[:, :, -1:] + self.eps)
+        output = torch.reshape(output, (batch_size, -1, height, width))
+
+        return output
+
+    def quadratic_attention(self, qkv: torch.Tensor) -> torch.Tensor:
+        batch_size, _, height, width = list(qkv.size())
+
+        qkv = torch.reshape(qkv, (batch_size, -1, 3 * self.attention_head_dim, height * width))
+        query, key, value = (
+            qkv[:, :, 0 : self.attention_head_dim],
+            qkv[:, :, self.attention_head_dim : 2 * self.attention_head_dim],
+            qkv[:, :, 2 * self.attention_head_dim :],
+        )
+
+        query = self.kernel_nonlinearity(query)
+        key = self.kernel_nonlinearity(key)
+
+        scores = torch.matmul(key.transpose(-1, -2), query)
+
+        original_dtype = scores.dtype
+        scores = scores.float()
+        scores = scores / (torch.sum(scores, dim=2, keepdim=True) + self.eps)
+        scores = scores.to(original_dtype)
+
+        output = torch.matmul(value, scores)
+        output = torch.reshape(output, (batch_size, -1, height, width))
+
+        return output
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        residual = hidden_states
+
+        qkv = self.to_qkv(hidden_states)
+
+        multi_scale_qkv = [qkv]
+        for block in self.to_qkv_multiscale:
+            multi_scale_qkv.append(block(qkv))
+
+        qkv = torch.cat(multi_scale_qkv, dim=1)
+
+        height, width = qkv.shape[-2:]
+        if height * width > self.attention_head_dim:
+            hidden_states = self.linear_attention(qkv).to(qkv.dtype)
+        else:
+            hidden_states = self.quadratic_attention(qkv)
+
+        hidden_states = self.proj_out(hidden_states)
+        hidden_states = self.norm_out(hidden_states)
+
+        return hidden_states + residual
+
+
+class EfficientViTBlock(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        heads_ratio: float = 1.0,
+        dim: int = 32,
+        qkv_multiscales: Tuple[int, ...] = (5,),
+        norm_type: str = "batch_norm",
+    ):
+        super().__init__()
+
+        self.attn = LiteMLA(
+            in_channels=in_channels,
+            out_channels=in_channels,
+            heads_ratio=heads_ratio,
+            attention_head_dim=dim,
+            norm_type=norm_type,
+            kernel_sizes=qkv_multiscales,
+        )
+
+        self.conv_out = GLUMBConv(
+            in_channels=in_channels,
+            out_channels=in_channels,
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.attn(x)
+        x = self.conv_out(x)
+        return x
+
+
+def get_block_from_block_type(
```

Comment:
```suggestion
def get_block(
```
to be consistent with similar methods from other models